lib/Target/HSACO/HSACOTranslation.cpp

```diff
@@ -144,7 +146,33 @@ std::string generate_hsaco(llvm::Module *module, const std::string &triple,
   std::string hsaco_path = kernel_name + std::string(".hsaco");
 
   std::string error_message;
-  std::string lld_path = ::triton::tools::getenv("ROCM_PATH") + "/llvm/bin/ld.lld";
+  std::string rocm_path = ::triton::tools::getenv("ROCM_PATH");
+  std::string lld_suffix = "/llvm/bin/ld.lld";
+  std::string lld_path;
+  if (rocm_path.empty()){
+    std::cout << "ROCM_PATH Not found...Trying /opt/rocm" << std::endl;
+    rocm_path = "/opt/rocm";
+  } 
+
+  lld_path = "/opt/rocm" + lld_suffix;
```

Comment:
Maybe you want something like this?
```suggestion
  lld_path = rocm_path + lld_suffix;
```